src/ModelContextProtocol.Core/Client/StdioClientTransport.cs

```diff
@@ -159,15 +165,20 @@ public async Task<ITransport> ConnectAsync(CancellationToken cancellationToken =
 #if NET
             processStarted = process.Start();
 #else
-            Encoding originalInputEncoding = Console.InputEncoding;
-            try
-            {
-                Console.InputEncoding = StreamClientSessionTransport.NoBomUtf8Encoding;
-                processStarted = process.Start();
-            }
-            finally
+            // IMPORTANT: This must be synchronized to prevent race conditions when multiple
+            // transports are created concurrently.
+            lock (s_consoleEncodingLock)
             {
-                Console.InputEncoding = originalInputEncoding;
+                Encoding originalInputEncoding = Console.InputEncoding;
+                try
+                {
+                    Console.InputEncoding = StreamClientSessionTransport.NoBomUtf8Encoding;
```

Comment:
Could this still race with other code in the same process changing `Console.InputEncoding`?  We synchronized our access, but this property can be changed by anything in process.